.travis.yml

```diff
@@ -11,12 +11,16 @@ matrix:
 install:
 - |
   if [[ ${MASON_PLATFORM:-unset} == 'unset' ]] || [[ ${MASON_PLATFORM} == 'osx' ]] || [[ ${MASON_PLATFORM} == 'linux' ]]; then
-    echo "Installing clang for OSX/Linux via 'install' target"
-    source utils/toolchain.sh
-    which ${CC}
-    ${CC} -v
-    which ${CXX}
-    ${CXX} -v
+    # If platform is Linux or OSX, we might still be cross compiling if the
+    # MASON_PLATFORM_VERSION is not the same as the host platform.
+    if [[ ${MASON_PLATFORM_VERSION} == `uname -m` ]] || [[ -z ${MASON_PLATFORM_VERSION} ]]; then
```

Comment:
This also needs to skip android and ios builds. So I think we need to keep the logic to only opt-in if `osx` or `linux` is the `MASON_PLATFORM` otherwise this will incorrectly trigger installing this clang++ version for those platforms where we want to use the platform specific compiler.